docs/VIEWS.md

```diff
@@ -0,0 +1,168 @@
+    </main>
+```
+which will hold all the content of our `IndexView` (*and here we begin the introduction of [BEM](http://getbem.com/naming/)-naming*).
+
+---
+
+Next is our first tag that is different with different props
+```typescript
+import Loader from '../../components/Loader';
+// ...
+    <main className="index">
+        {loading && <Loader />}
+    </main>
+```
+which means, that depending if the **prop** `loading` is true (*using [logical operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators)-shorthand that returns the right hand-side if the both sides evaluate as true*), we want to **render** the `Loader`-component we defined earlier in the [Components](/COMPONENTS.md) section.
+
+---
+
+Next up we define the title and a form to add a new `Todo`
+```typescript
+    <main className="index">
+    // ...
+
+        <h1 className="index__header">Todo app</h1>
+        <form className="index__form" onSubmit={e => e.preventDefault()}>
+            <label className="index__form__label" htmlFor="newtodo">Add a new todo:</label>
+            <input
+                className="index__form__input"
+                name="newtodo"
+                type="text"
+                autoFocus={true}
+                value={title}
+                onChange={e => setTitle(e.target.value)}
+            />
+            <Button click={saveTodo} text="Add" />
+        </form>
+    </main>
+```
+where we first define the title (*in this case a static string `Todo app`*). Next we define a form to create new `Todos` which introduces us to a lot of nice features of **React**, but first we do a little *"hack"* as we want to enable the user to press the `enter`-key when submitting a new `Todo` but we don't want to send the form to a new page, we set the value of `onSubmit` (*the handler for a form's submit method in React*) as `e => e.preventDefault()` where `e` is the event received and it's function `.preventDefault()` will (*as its name implies*) prevent the default functionality (*in this case send the form data to the current url, causing a reload*).
+
+Next we define a label for the `input`-field utilizing **React**'s `htmlFor`-value which is an alias for the [for](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label)-attribute. Then we define the actual `input`-field for inputting a `Todo`, which is a simple text `input`, but we also make it [autofocus](https://developer.mozilla.org/en/docs/Web/HTML/Element/input) (*if you open the page, you can start typing into it directly*) and make it a [controlled component](https://facebook.github.io/react/docs/forms.html#controlled-components), meaning that whenever the value of `title` changes, the `input` will also update. We also add the `onChange`-listener to actually set the new `title` the user has typed in. Finally we create a simple `Button` (*we defined earlier in [components](/COMPONENTS.md#button)*) to submit the form.
```

Comment:
So `htmlFor` is not just [the DOM API property](https://developer.mozilla.org/en-US/docs/Web/API/HTMLLabelElement/htmlFor)?